### PR TITLE
nix_ci_builder: forward NIX_CI_SSH_KEY on the nix path

### PR DESCRIPTION
**Pull request by yorickvP**
https://github.com/serokell/serokell-ops/pull/131

### CODE REVIEW

File: src/nix_ci_builder.erl [hunk 1]
```diff
@@ -20,7 +20,18 @@ consume_port(Port, Output) ->
 	    {Status, Output}
 	end.
 
+ensure_key(false) ->
+    true.
+
+ensure_key(Key) ->
+    Outpath = <<"/tmp/nix-build-ssh-key">>,
```
Comment (lukateras) [inline src/nix_ci_builder.erl:27]: That is dangerous outside of systemd `PrivateTmp` hack.

File: src/nix_ci_builder.erl [hunk 2]
```diff
@@ -20,7 +20,18 @@ consume_port(Port, Output) ->
 	    {Status, Output}
 	end.
 
+ensure_key(false) ->
+    true.
```
Comment (lukateras) [inline src/nix_ci_builder.erl:24]: Usually, `ok` is used in this case.

File: src/nix_ci_builder.erl [hunk 3]
```diff
@@ -20,7 +20,18 @@ consume_port(Port, Output) ->
...
+
+ensure_key(Key) ->
+    Outpath = <<"/tmp/nix-build-ssh-key">>,
+    file:copy(Key, Outpath),
+    file:change_group(Outpath, 30000), % nixbld, but no getgrnam
```
Comment (lukateras) [inline src/nix_ci_builder.erl:29]: I don't think Nix enforces this gid, so this is probably less than portable :-(

File: src/nix_ci_builder.erl [hunk 4]
```diff
@@ -20,7 +20,18 @@ consume_port(Port, Output) ->
...
+    file:change_mode(Outpath, 8#600),
+    os:putenv(<<"NIX_PATH">>, io_lib:format(<<"ssh-key=~s">>, [Outpath])).
```
Comment (lukateras) [inline src/nix_ci_builder.erl:31]: This rewrites `nixpkgs` on the NIX_PATH.

**Review by lukateras — COMMENTED** [review]
This approach seems moderately flaky. I think it would be better to leave `NIX_PATH` management to NixOS configuration on the CI server.

File: src/nix_ci_builder.erl (hunk 4 above)
Comment (yorickvP) [reply src/nix_ci_builder.erl:31]: Is that wrong?

File: src/nix_ci_builder.erl (hunk 3 above)
Comment (yorickvP) [reply src/nix_ci_builder.erl:29]: It's hardcoded in nixos in multiple places, but not in Nix

File: src/nix_ci_builder.erl (hunk 1 above)
Comment (yorickvP) [reply src/nix_ci_builder.erl:27]: It's supposed to be readable by the nixbld user. Probably needs a different mode though (040)?

File: src/nix_ci_builder.erl (hunk 4 above)
Comment (lukateras) [reply src/nix_ci_builder.erl:31]: Yes, we should support use case where people just want to use `with import <nixpkgs> {};` in their `default.nix` without pinning nixpkgs.

**Comment by yorickvP** [comment]
See https://github.com/serokell/serokell-ops/pull/131 for the alternative.

**Pull request closed without merging**